wdl-parsing/WDLParsingFunction/src/main/java/io/dockstore/wdlparser/App.java

```diff
@@ -0,0 +1,177 @@
+import java.util.Map;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.openapitools.client.model.LanguageParsingRequest;
+import org.openapitools.client.model.LanguageParsingResponse;
+import org.openapitools.client.model.VersionTypeValidation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.JavaConverters;
+import womtool.WomtoolMain;
+
+/** Handler for requests to Lambda function. */
+public class App
+    implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+  ObjectMapper mapper = new ObjectMapper();
+  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
+
+  @Override
+  public APIGatewayProxyResponseEvent handleRequest(
+      final APIGatewayProxyRequestEvent input, final Context context) {
+
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", "application/json");
+    headers.put("X-Custom-Header", "application/json");
+
+    APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+    if (input != null && input.getBody() != null) {
+      try {
+        LanguageParsingRequest request =
+            mapper.readValue(input.getBody(), LanguageParsingRequest.class);
+        try {
+          String s =
+              parseWdlFile(
+                  request.getUri(),
+                  request.getBranch(),
+                  request.getDescriptorRelativePathInGit(),
+                  request);
+          return response.withStatusCode(HttpURLConnection.HTTP_OK).withBody(s);
+        } catch (IOException e) {
+          String errorMessage = "Could not clone repository to temporary directory";
+          LOGGER.error(errorMessage, e);
+          return response
+              .withBody(errorMessage)
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        } catch (GitAPIException e) {
+          StringWriter sw = new StringWriter();
+          e.printStackTrace(new PrintWriter(sw));
+          String exceptionAsString = sw.toString();
+          return response
+              .withBody(exceptionAsString)
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        }
+      } catch (IOException e) {
+        String errorMessage = "Could not process request";
+        LOGGER.error(errorMessage, e);
+        return response.withBody(errorMessage).withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+      }
+    } else {
+      return response
+          .withBody("No body in request")
+          .withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+    }
+  }
+
+  private String parseWdlFile(
+      String uri,
+      String branch,
+      String descriptorRelativePathInGit,
+      LanguageParsingRequest languageParsingRequest)
+      throws IOException, GitAPIException {
+    Path tempDirWithPrefix = Files.createTempDirectory("clonedRepository");
```

Comment:
Does this need to be deleted?

https://stackoverflow.com/questions/44108712/aws-lambda-release-tmp-storage-after-each-execution, which is a little old, says the disk can get reused.